scipy/stats/stats.py

```diff
@@ -8122,6 +8122,12 @@ def rankdata(a, method='average', *, axis=None):
     axis : {None, int}, optional
         Axis along which to perform the ranking. If ``None``, the data array
         is first flattened.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan. 'propagate' just execute
+        the function without checking for nan, 'raise' throws an error,
+        'omit' performs the calculations ignoring nan values.
+        Default is 'propagate'.  Note that when the value is 'omit',
+        the input after removing nans needs not to be ragged shape.
```

Comment:
I don't think we want to add that restriction if it weren't the case before.

Reply:
The [doc](http://scipy.github.io/devdocs/dev/api-dev/nan_policy.html) says "nan_policy='omit': Ignore occurrences of nan in the input". I understand the proper behavior is 

```
x = np.array([[1, np.nan], [3, 4]])
print(rankdata(x, axis=1, nan_poliby="omit"))
# array([[1], [2, 1]]) <- ragged array
```
Do you think is it OK the output can include nan when "omit" is used?

Reply:
Take as look at what gh-13166 does. I interpret the docstring to mean that it includes nans in the output, but they do not affect the calculation.